pkg/snapshot/snapshot.go

```diff
@@ -402,11 +407,20 @@ func writePolicy(idx int, userPolicy *tykv1alpha1.SecurityPolicySpec, w *bufio.W
 	}
 
 	pol.Spec = *userPolicy
-	pol.Spec.ID = userPolicy.MID
-	pol.Spec.OrgID = ""
+
+	pol.Spec.ID = new(string)
+	if userPolicy.MID != nil {
+		*pol.Spec.ID = *userPolicy.MID
+	}
+
+	if pol.Spec.OrgID == nil {
+		pol.Spec.OrgID = new(string)
+	}
+
+	*pol.Spec.OrgID = ""
 
 	for i := 0; i < len(pol.Spec.AccessRightsArray); i++ {
-		apiID := pol.Spec.AccessRightsArray[i].APIID
+		apiID := *pol.Spec.AccessRightsArray[i].APIID
```

Comment:
what do you think checking `pol.Spec.AccessRightsArray[i]` is nil or not before accessing APIID?